src/sentry/models/notificationsetting.py

```diff
@@ -0,0 +1,102 @@
+from enum import Enum
+
+from django.conf import settings
+
+from sentry.db.models import (
+    BoundedPositiveIntegerField,
+    FlexibleForeignKey,
+    Model,
+    sane_repr,
+)
+
+
+class NotificationSettingTypes(Enum):
+    DEFAULT = 0
+    DEPLOY = 10
+    ISSUE_ALERTS = 20
+    WORKFLOW = 30
+
+
+NOTIFICATION_SETTING_TYPES = {
+    NotificationSettingTypes.DEFAULT: "default",
+    NotificationSettingTypes.DEPLOY: "deploy",
+    NotificationSettingTypes.ISSUE_ALERTS: "issue",
+    NotificationSettingTypes.WORKFLOW: "workflow",
+}
+
+
+class NotificationSettingOptionValues(Enum):
+    DEFAULT = 0  # Defer to a setting one level up.
+    NEVER = 10
+    ALWAYS = 20
+    SUBSCRIBE_ONLY = 30  # workflow
+    COMMITTED_ONLY = 40  # deploy
+
+
+NOTIFICATION_SETTING_OPTION_VALUES = {
+    NotificationSettingOptionValues.DEFAULT: "default",
+    NotificationSettingOptionValues.NEVER: "off",
+    NotificationSettingOptionValues.ALWAYS: "on",
+    NotificationSettingOptionValues.SUBSCRIBE_ONLY: "subscribe_only",
+    NotificationSettingOptionValues.COMMITTED_ONLY: "committed_only",
+}
+
+
+class NotificationSettingProvider(Enum):
+    EMAIL = 0
+    SLACK = 1
+
+
+NOTIFICATION_SETTING_PROVIDER = {
+    NotificationSettingProvider.EMAIL: "email",
+    NotificationSettingProvider.SLACK: "slack",
+}
+
+
+class NotificationSetting(Model):
+    """A setting of when to notify a user or team about activity within the app"""
+
+    __core__ = False
+
+    organization = FlexibleForeignKey("sentry.Organization", null=True)
+    project = FlexibleForeignKey("sentry.Project", null=True)
```

Comment:
`project` is one of our dangerous tables, as in it can be hard to add a postgres level fk here. Set `db_constraint=True` for this, so that way we get to still have the relation in Django, but we don't have the constraint at the postgres level.